src/codegen/compile.ml

```diff
@@ -5794,11 +5794,22 @@ module FuncDec = struct
       compile_unboxed_const (E.add_fun_ptr env (E.built_in env reject_name)) ^^
       get_cb_index
 
-  let ignoring_callback env =
+  let _ignoring_callback env =
     let name = "@ignore_callback" in
     Func.define_built_in env name ["env", I32Type] [] (fun env -> G.nop);
     compile_unboxed_const (E.add_fun_ptr env (E.built_in env name))
 
+  let _faulting_success_callback env =
+    let name = "@faulting_callback" in
```

Comment:
You've used the same name for the faulting and success callback.

Reply:
That may be your bug...

Reply:
Yeah, fixing that gives me
```
oneway-cleanup: [tc] [comp] [comp-ref] [valid] [valid-ref] [ic-ref-run]
--- oneway-cleanup.ic-ref-run (expected)
+++ oneway-cleanup.ic-ref-run (actual)
@@ -0,0 +1,11 @@
+→ update create_canister(record {dnczaeh = null})
+← replied: (record {hymijyo = principal "cvccv-qqaaq-aaaaa-aaaaa-c"})
+→ update install_code(record {arg = blob ""; kca_xin = blob "\00asm\01\00\00\00\0…
+← replied: ()
+→ update go()
+debug.print: oneway_success!
+debug.print: (success) ABOUT TO FAULT
+debug.print: oneway_fail!
+debug.print: canister trapped: EvalTrapError region:0xXXX-0xXXX "canister trapped explicitly: assertion failed at oneway-cleanup.mo:11.5-11.17"
+debug.print: (rejection) ABOUT TO FAULT
+← replied: ()
```
which is perfect! Damn copy&paste mistakes... Thanks for looking.